tests/logger.cpp

```diff
@@ -0,0 +1,14 @@
+#include "stream-client/logger.hpp"
+
+int main()
+{
+    const auto used_level = stream_client::log_level::debug;
+    auto logger = std::make_shared<stream_client::cout_logger>(used_level);
+    stream_client::set_logger(used_level, [logger](stream_client::log_level level, const std::string& location, const std::string& message) {
+        logger->message(level, location, message);
+    });
+    STREAM_LOG_TRACE("just a trace");
+    STREAM_LOG_ERROR("funny error");
+    STREAM_LOG_DEBUG("debug message from bunny");
+    return 0;
```

Comment:
The purpose of tests is to check and anchor public API. Meaning that the behavior and signature of any user-available function or implemented public class methods should be called, tested, and asserted for proper results. This code does not test anything, if we change log functions to do nothing this will return success. If we refactor `log_interface` completely, this will succeed, meaning that we are not using testing to our advantage.

There should be test-cases at least checking that:
- `func_logger::message` calls passed callback
- `cout_logger::message` prints messages to stdout
- `set_logger` does sets a global logger
- `set_log_level`/`get_log_level` operate on global logger level
- `log_message` calls global logger `message` method